test/battle/hold_effect/eject_pack.c

```diff
@@ -63,3 +63,27 @@ SINGLE_BATTLE_TEST("Eject Pack is triggered by self-inflicting stat decreases")
         ANIMATION(ANIM_TYPE_MOVE, MOVE_CELEBRATE, opponent);
     }
 }
+
+DOUBLE_BATTLE_TEST("Eject Pack activation will not trigger an attack from the incoming mon")
+{
+    GIVEN {
+        PLAYER(SPECIES_TATSUGIRI) { Speed(10); Ability(ABILITY_COMMANDER); }
+        PLAYER(SPECIES_WOBBUFFET) { Speed(100); Item(ITEM_EJECT_BUTTON); }
+        PLAYER(SPECIES_DONDOZO) { Speed(20); }
+        OPPONENT(SPECIES_WOBBUFFET) { Speed(50); Item(ITEM_EJECT_PACK); }
+        OPPONENT(SPECIES_WOBBUFFET) { Speed(10); }
+        OPPONENT(SPECIES_WYNAUT) { Speed(1); }
+    } WHEN {
+        TURN { MOVE(opponentRight, MOVE_MAKE_IT_RAIN); SEND_OUT(playerRight, 2); SEND_OUT(opponentRight, 2); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_MAKE_IT_RAIN, opponentRight);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_HELD_ITEM_EFFECT, playerRight);
+        MESSAGE("Wobbuffet is switched out with the Eject Button!");
+        ABILITY_POPUP(playerLeft, ABILITY_COMMANDER);
+        MESSAGE("Tatsugiri was swallowed by Dondozo and became Dondozo's commander!");
+        NONE_OF {
+            ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_HELD_ITEM_EFFECT, opponentLeft);
+            MESSAGE("Wobbuffet is switched out with the Eject Pack!");
```

Comment:
If possible, it would be best to test that neither the ejected mon's slot or Tatsugiri's slot use a move (Celebrate I assume they would try to use)

Reply:
They didn't use any move. Just attacked with a corrupted move.

Reply:
It would be best for this to watch the recording but the test is illegal without the fix. 

Also I moved it to the eject button file